src/main.cpp

```diff
@@ -89,88 +89,80 @@ void setupPHYSFS()
-					fs.setWriteDir(baseSearchPath + "../");
-					fs.probeDir("Documents");
-				#endif
-				#if TARGET_OS_IPHONE
-					std::string userdir = baseSearchPath + "../Documents/";
-				#else
-					std::string userdir = fs.getUserDir();
-				#endif
-			#else
-				// Linux
-				std::string userdir = fs.getUserDir();
-
-				// handle the case when it is installed
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby");
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/gfx.zip");
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/sounds.zip");
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/scripts.zip");
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/backgrounds.zip");
-				fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/rules.zip");
-			#endif
-			std::string userAppend = ".blobby";
-			std::string homedir = userdir + userAppend;
-			/// \todo please review this code and determine if we really need to add userdir to search path
-			/// only to remove it later
-			fs.setWriteDir(userdir);
-			fs.probeDir(userAppend);
-			/// \todo why do we need separator here?
-			fs.probeDir(userAppend + separator + "replays");
-			fs.probeDir(userAppend + separator + "gfx");
-			fs.probeDir(userAppend + separator + "sounds");
-			fs.probeDir(userAppend + separator + "scripts");
-			fs.probeDir(userAppend + separator + "backgrounds");
-			fs.probeDir(userAppend + separator + "rules");
-			fs.removeFromSearchPath(userdir);
-			// here we set the write dir anew!
-			fs.setWriteDir(homedir);
-			#if defined(GAMEDATADIR)
-			{
-				// A global installation path makes only sense on non-Windows
-				// platforms
-				std::string basedir = GAMEDATADIR;
-				fs.addToSearchPath(basedir);
-				fs.addToSearchPath(basedir + separator + "gfx.zip");
-				fs.addToSearchPath(basedir + separator + "sounds.zip");
-				fs.addToSearchPath(basedir + separator + "scripts.zip");
-				fs.addToSearchPath(basedir + separator + "backgrounds.zip");
-				fs.addToSearchPath(basedir + separator + "rules.zip");
-			}
-			#endif
-		#else
-			fs.setWriteDir(baseSearchPath);
-			fs.probeDir("replays");
-			fs.probeDir("gfx");
-			fs.probeDir("sounds");
-			fs.probeDir("scripts");
-			fs.probeDir("backgrounds");
-			fs.probeDir("rules");
-		#endif
+	/*
+	set read dir (unix-like when installed)
+	*/
+	#if BLOBBY_ON_DESKTOP && (!defined WIN32)
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby");
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/gfx.zip");
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/sounds.zip");
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/scripts.zip");
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/backgrounds.zip");
+		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby/rules.zip");
+	#endif
 
+	/// \todo the following lines must be refactored!!!!
+	#if (!defined __ANDROID__) && !(defined __APPLE__ && TARGET_OS_IPHONE) && (!BLOBBY_ON_DESKTOP)
```

Comment:
this condition is somewhat hard to understand. If we're not on android, desktop, or ios, does that leave anything but the switch?

Reply:
In the best case this part is completely obsolet.
It is only there for default write dir handling previously used (which includes switch).
I added an additional branch which changes default and switch write dir behavior.
@SilentFlyBy : Can you please check if **fix/path-handling-switch** is working for switch?

Reply:
I merged **fix/path-handling-switch** into this PR. If we get any problems with switch we can fix it later.